src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +45,64 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
     return false;
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.PROFILE);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
     return false;
   }
-  async function updateUser() {
-    const payload = getPayload();
 
+  async function handleProfileSuccess() {
+    await updateUser({ ...getPayload() }, () => setActiveTab(Tabs.SUCCESS));
+  }
```

Comment:
So my comments might be out of order, was reading and trying to follow the logic and renaming changes.

I think this would make more sense to be called `handleUpdateSubmit` or something. because the responsibility of this is to invoke the `updateUser` method, and not actually do on a `ProfileSuccess`

Also might be foggy on the async await, maybe @qdozaq or @mat10tng can confirm, but you wrote updateUser as an async function that's actually invoking the call to the API with User.update. I'm not sure we need async declared for this function.

Reply:
I'm about to push my changes. To use await you need async declared, so this is the one thing I didn't change. If you want though I could switch it to .then().

Reply:
very happy to stick with await fwiw.. 

await vs .then() is a style question honestly